File: Programmers/Level 2/소수 찾기.js
const powerset = (array) => {
  const result = [[]]
  for (let value of array) {
    const length = result.length
    for (let i=0; i<length; i++){
      let temp = result[i].slice(0)
      temp.push(value)
      result.push(temp)
    }
  }
  return result;
}

const permutation = (array) => {
    return array.reduce((arr, ele) => {
        const result = []
        arr.forEach((a) => {
            for (let i=a.length; i>=0; i--) {
                const newa = [].concat(a)
                newa.splice(i, 0, ele)
                result.push(newa)
            }
        })
        return result
    }, [[]]
    )
}

function solution(numbers) {
    let newNumbers = powerset(numbers).splice(1).map((a) => a.join('')).reduce((a, b) =>
        a.includes(b) ? a : [...a, b]
    , [])
    let answer = []
    for (let number of newNumbers) {
        const num = permutation(number.split(''))
        for (let n of num) {
            n = parseInt(n.join(''))
            let flag = true
            for (let i=2; i<n; i++) {
                if ((n % i) === 0) {
                    flag = false
                    break
                }
            }
            if (n > 1 && flag) answer.push(n)
        }
    }
    answer = [...new Set(answer)]
    return answer.length
}

File: Programmers/Level 2/소수 찾기.py
import itertools


def solution(numbers):
    answer = 0
    numbers = list(sum(map(lambda x: x.split(), numbers), []))

    _combs = []
    for i in range(2, len(numbers)+1):
        temp = list(itertools.combinations(numbers, i))
        _combs += temp

    _set = []
    for _comb in _combs:
        temp = list(map(list, list(itertools.permutations(_comb))))
        _set += temp
    _set = list(map(''.join, _set))

    numbers += _set
    numbers = list(set(map(int, numbers)))
    for number in numbers:
        if number < 2:
            continue
        for i in range(2, number):
            if number % i == 0:
                break
        else:
            answer += 1

    return answer


# print(solution("17"))  # 3
# print(solution("011"))  # 2
